### PR TITLE
MVC for Skill database.

### PR DESCRIPTION
**Pull request by talha8045**

### CODE REVIEW

**Review by flozender — CHANGES_REQUESTED** [review]
Please make the changes

File: Client/index.js [hunk 1]
```diff
@@ -53,64 +53,136 @@ app.get("/about", async (req, res) => {
...
-      "CodeChef (CC)",
-      "Codeforces (CF)",
-      "InterviewBit (IB)",
-      "Spoj (S)",
-      "Geeks For Geeks (GFG)",
-      "BuildIT",
-      "Overall Score",
-      "Weekly Performance",
-      "Points",
-    ];
```
[the remark this answers is not in the document]
Comment (talha8045) [reply Client/index.js:74]: Done. I've added headers loop back.

File: Client/views/skilladd.ejs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (talha8045) [reply Client/views/skilladd.ejs]: Done. lastly i have to replace 'weekId: String' with 'weekId: Date()'.

**Comment by talha8045** [comment]
Done. i committed all the chages you suggested.

**Comment by flozender** [comment]
Let's sort this to the latest week on top followed by the other in the order. And if a CSV is uploaded for a week that already has a record, update it, it's creating multiple copies right now. 
image

**Comment by flozender** [comment]
On selecting a previous week, let's make the date show up here 
image

File: Client/views/skilladd.ejs (hunk 2 above)
[the remark this answers is not in the document]
Comment (flozender) [reply Client/views/skilladd.ejs]: No, `weekId: String` is right, since it's still an ID that happens to be in the format of a date.

**Comment by flozender** [comment]
Since you added a new ejs file, remove `uploadcsv.ejs` they both do the same thing.

File: Client/index.js [hunk 3]
```diff
@@ -238,7 +368,7 @@ app.get("/admin/add/pdf", async (req, res) => {
 app.get("/admin/add/leaderboard", async (req, res) => {
   let url = {
     url: clientRoute,
-    serverurl: serverRoute,
+    serverurl: serverRoute + "/updateLeaderboard",
```
Comment (flozender) [inline Client/index.js:371]: I think we don't need this anymore right? Since you have your `/skillExcel` endpoint for leaderboard updation?

**Review by flozender — CHANGES_REQUESTED** [review]

File: Client/index.js (hunk 3 above)
Comment (talha8045) [reply Client/index.js:371]: I shall remove it.

File: Client/index.js [hunk 4]
```diff
@@ -52,65 +52,194 @@ app.get("/home", async (req, res) => {
 app.get("/about", async (req, res) => {
   res.render("about", { imgUsername: req.cookies.username });
 });
+
 app.get("/skillup365", async (req, res) => {
```
Comment (flozender) [inline Client/index.js:56]: Why do we have an `app.get("/skillup365",)` and an `app.get("/skill",)`? Remove the one you're not using.

File: Client/views/leaderboard.ejs [hunk 5]
```diff
@@ -147,6 +177,21 @@
           </div>
           <% } %>
         </div>
+      </div>
+    </section>
+    <div>
+      <form action="<%= data.clientAddress %>/skill/" method="POST">
+        <select class="dropdown" name="weekId" onchange="javascript:$('#submit').click();">
```
Comment (flozender) [inline Client/views/leaderboard.ejs:184]:
```suggestion
        <select class="dropdown custom-select" name="weekId" onchange="javascript:$('#submit').click();">
```

**Review by flozender — CHANGES_REQUESTED** [review]
Final review before we merge

File: Client/index.js (hunk 4 above)
Comment (talha8045) [reply Client/index.js]: It is done.

File: Client/views/leaderboard.ejs (hunk 5 above)
Comment (talha8045) [reply Client/views/leaderboard.ejs]: Okay

File: Client/views/leaderboard.ejs [hunk 6]
```diff
@@ -114,8 +124,12 @@
...
           style="padding-bottom: 0.5rem"
         >
-          <h2>This Week's Toppers</h2>
-          <a href="../images/SkillUP.pdf">
+          <% if( data[0].weekId !== week[ week.length-1 ].weekId ) { %>
+            <h2>Toppers for  <%= data[0].weekId.substring(8,10) %>-<%= data[0].weekId.substring(5,7) %>-<%= data[0].weekId.substring(0,4) %></h2>
+          <% } else { %>
+            <h2>This Week's Toppers</h2>
+          <% } %>
+          <a href="../images/SkillUp.pdf">
```
Comment (flozender) [inline Client/views/leaderboard.ejs:132]:
```suggestion
          <a href="../images/SkillUP.pdf">
```

**Review by flozender — CHANGES_REQUESTED** [review]

File: Client/views/uploadcsv.ejs [hunk 7]
```diff
@@ -18,31 +18,53 @@
     <link rel="stylesheet" href="../../css/navadmin.css" />
     <link rel="icon" type="image/png" href="../../images/logo-backup2.png" />
     <script src="/adminjs/modernizr.custom.js"></script>
+    <style>
+			[type="date"]::-webkit-calendar-picker-indicator {
+				opacity: 0;
+			}
+			
+			#date {
+				background:#fff url(https://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/calendar_2.png)  97% 50% no-repeat;
```
Comment (flozender) [inline Client/views/uploadcsv.ejs:27]: Save this image locally and link it here, the URL may change in the future, which will break the UI.

Comment (talha8045) [reply Client/views/uploadcsv.ejs]: The image is now saved locally.

**Comment by flozender** [comment]
Thanks for the great PR!

**Pull request merged**